src/Form/AuthenticationSettingsForm.php

```diff
@@ -184,17 +184,28 @@ protected function getUserConditionDefinitions(): array {
     $definitions = $this->conditionManager->getFilteredDefinitions('oe_authentication_2fa', [$context]);
     // The ::getFilteredDefinitions() call above filters out plugins that match
     // the context definition, as well as plugins without any context definition
-    // required. We need to filter those out manually.
-    return array_filter($definitions, function ($definition) {
+    // required, and plugins with optional context definitions.
+    // We need to filter those out manually, and return only conditions that
+    // require one single user context.
+    return array_filter($definitions, function ($definition): bool {
       // @see \Drupal\Core\Plugin\Context\ContextHandler::getContextDefinitions()
       if ($definition instanceof ContextAwarePluginDefinitionInterface) {
-        return !empty($definition->getContextDefinitions());
+        $context_definitions = $definition->getContextDefinitions();
       }
-      if (is_array($definition) && isset($definition['context_definitions'])) {
-        return !empty($definition['context_definitions']);
+      elseif (is_array($definition) && isset($definition['context_definitions'])) {
+        $context_definitions = $definition['context_definitions'];
+      }
+      else {
+        return FALSE;
+      }
+
+      if (empty($context_definitions) || count($context_definitions) > 1) {
+        return FALSE;
       }
 
-      return FALSE;
+      // At this point the single context definition available is a user entity
+      // one. If it's required, the plugin is allowed.
+      return current($context_definitions)->isRequired();
     });
```

Comment:
So to be clear:\
We are filtering conditions in the form, but we don't filter them when they are applied, or the config is stored.\
This means they could be enabled by hacking the config yml and then importing.

I don't think it is a problem, site builders who edit config files do at their own risk.\
Also it would be out of scope to change it here.\
I only comment because I think it is the first time I look at this code (or maybe I don't remember it).

Reply:
Yes config can be manually changed, but that's the same everywhere.